mlserver/batch_processing.py

```diff
@@ -101,17 +102,21 @@ def from_inference_request(
         cls, inference_request: InferenceRequest, binary_data: bool
     ) -> "TritonRequest":
         inputs = []
-
         for request_input in inference_request.inputs or []:
             new_input = httpclient.InferInput(
                 request_input.name, request_input.shape, request_input.datatype
             )
+            request_input_np = NumpyCodec.decode_input(request_input)
+
+            # Change datatype if BYTES to satisfy Tritonclient checks
```

Comment:
I have the feeling this should be something handled in the `NumpyCodec` but it may be too `tritonclient` specific...

Reply:
Good point - do you know what's returned by `NumpyCodec.decode_input` at the moment?

Reply:
I think what's going is that the NumpyCodec encodes that list as a tensor of strings (i.e. using `np.dtype(str)`). However, Triton doesn't know how to handle that case. 

https://github.com/SeldonIO/MLServer/blob/d283fc0c6f07f731ed94139847da9e70d9be2497/mlserver/codecs/numpy.py#L41-L43

I can't remember why we had to add that `np.dtype(str)` special case, but I would be keen to keep it as is to avoid any potential side effects. Having said that, it's unclear whether changing the `dtype` here may cause any issues downstream (although I'd expect that strings in np arrays are quite an edge case).